### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by JuanDBta**
In this branch, I created the remaining classes for our school library and created the associations between them.
 You can find the following features:

Created a class **Classroom** with the following:

  - **@label** instance variable, should be initialized in the constructor.
  - Setter and getter for **@label**

Created the has-many/belongs-to relationship between **Classroom** and **Student**. The following should be implemented:

  - Created the has-many side (a classroom has many students).
  - Created the belongs-to side (a student belongs to a classroom).

Created a class **Book** with the following:

  - **@title** and **@author** instance variables, should be initialized in the constructor.
  - Setters and getters for instance variables.

Created a class **Rental** with the following:

  - **@date** instance variable, should be initialized in the constructor.
  - Setter and getter for **@date**.

Created the many-to-many (also has-many-through) relationship between **Person** and **Book** using the intermediate class **Rental**. The following was implemented:

  - Created the has-many side of **Book** and **Rental** (a book has many rentals).
  - Created the belongs-to side of **Rental** and **Book** (a rental belongs to a book).
  - Created the has-many side of **Person** and **Rental** (a person has many rentals).
  - Created the belongs-to side of **Rental** and **Person** (a rental belongs to a person).
  - Modified the constructor of **Rental** so **Book** and **Person** are set in it.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    @rentals.push(rental)
+  end
```
Comment (zieeco) [inline book.rb]: - Hey, the add rental method should have a date and the book itself should be sent as a parameter as this is required to create the new rental achieving the 'has-many' association. 
 please, update the method to be like this:

```ruby
 def add_rental(person, date)
    Rental.new(date, self, person)
  end
```

File: person.rb [hunk 2]
```diff
@@ -4,14 +4,15 @@
 
 class Person < Nameable
   attr_reader :id
-  attr_accessor :name, :age, :parent_permission
+  attr_accessor :name, :age, :parent_permission, :rentals
```
Comment (zieeco) [inline person.rb]: - You don't need the parent_permission here. Could you remove this as this should not be a setter/getter method?

File: person.rb [hunk 3]
```diff
@@ -4,14 +4,15 @@
...
 
   def initialize(age, name: 'Unknown', parent_permission: true)
     super()
     @id = Random.rand(1..1000)
     @name = name
     @age = age.to_i
     @parent_permission = parent_permission
```
Comment (zieeco) [inline person.rb:14]: - The same issue is applicable here!

File: person.rb [hunk 4]
```diff
@@ -29,6 +30,10 @@ def of_age?
   def can_use_services?
     of_age? || @parent_permission
   end
+
+  def add_rental(rental)
+    @rentals.push(rental)
+  end
```
Comment (zieeco) [inline person.rb]: - Hey, the add rental method should have a date and the person itself should be sent as a parameter to create the new rental achieving the 'has-many' association


```ruby
  def add_rental(book, date)
    Rental.new(date, book, self)
  end
```

File: rental.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+class Rental
+  attr_accessor :date, :person, :book
+
+  def initialize(date)
```
Comment (zieeco) [inline rental.rb]: - Hey, let's modify our constructor for Rental so Book and Person are set in it to break the many-to-many relationship. 
  `def initialize(date, book, person)`

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @JuanDBta,




Great job so far! 👏 
You've done well, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- No linter errors.✔️
- Nice adding the association. ✔️
- Descriptive PR. ✔️

# Required Changes ♻️
_Check the comments under the review._




## Optional suggestions
_Check the comments under the review._

**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by joskalenda — APPROVED** [review]
| STATUS: APPROVED ✅ |
|------------------------|
| Hi @JuanDBta , 🥰|
| Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: Congratulations! 🎉  |
|  |

### Highlights
✅ ... Clean code
✅ ... Readme file well organized

---------
### Optional suggestions

> _Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

### Cheers and Happy coding!👏👏👏
-------
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me(@joskalenda) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**